test/functional/test_framework/mininode.py

```diff
@@ -658,8 +658,6 @@ def on_inv(self, message):
                 # save txid
                 self.tx_invs_received[i.hash] += 1
 
-        super().on_inv(message)
-
     def get_invs(self):
         with mininode_lock:
```

Comment:
I'm a bit confused on why `get_invs` needs to grab the lock, but `on_inv` doesn't? I understand that in the case of `wait_for_broadcast`, `self.wait_until` already acquires the lock, but I just wanted to confirm that `on_inv` already expects to be holding the lock on call.

Reply:
Yes. All of the `on_<msgtype>()` already hold the lock, since they're called by the `on_message()` method (specifically, the `getattr(self, 'on_' + msgtype)(message)` line)